src/commands/another.ts

```diff
@@ -37,57 +37,65 @@ module.exports = new Command({
 
 		const guildMembers = await ia.guild.members.fetch()
 
-		lobby.tankPlayers.sort((a, b) => sortPlayers(a, b, lobby))
-		lobby.damagePlayers.sort((a, b) => sortPlayers(a, b, lobby))
-		lobby.supportPlayers.sort((a, b) => sortPlayers(a, b, lobby))
-
-		const topTanks = lobby.tankPlayers.slice(1)
-		const topDps = lobby.damagePlayers.slice(1)
-		const topSupports = lobby.supportPlayers.slice(1)
 		let playerMsg = ``
 		let btagMsg = ``
+		let foundPlayer: Signup | undefined
 
 		if (additionalRole === 'tank') {
-			topTanks.forEach(s => {
-				guildMembers.get(s.discordId)?.roles.add(ingameRole)
-				mongoSignups.updateOne({ discordId: s.discordId }, { $inc: { gamesPlayed: 1 } })
+			lobby.tankPlayers.sort((a, b) => sortPlayers(a, b, lobby))
+
+			// loop through players to find first player without ingame
+			lobby.tankPlayers.forEach(p => {
```

Comment:
Would replace .forEach with .find, as it exits, as soon as it finds one, instead of looping everyone